File: Go/binary_search.go
package main

import "fmt"

var vals []int = []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}

func main() {
	val := 3
	pos := binarySearch(val, 0, 9)
	fmt.Println(pos)

}

//recursive binary search
func binarySearch(val, low, high int) int {
	mid := int((low + high) / 2)
	if vals[mid] == val {
		return mid
	} else if vals[mid] > val {
		high = mid - 1
	} else {
		low = mid + 1
	}
	return binarySearch(val, low, high)
}

File: Python/binary_search.py
def binary_search(arr,item):
    begin = 0
    end = len(arr)-1
    while begin<=end:
        mid = begin +(end-begin)//2
        if arr[mid]==item:
            return "{} found at {}".format(item,mid+1)
        elif arr[mid]>item:
            end = mid-1
        else:
            begin = mid+1
    return "{} not found".format(item)
print(binary_search([5,90,234,567,100],567)) 
